src/autograd/Functions.cpp

```diff
@@ -147,6 +158,15 @@ namespace af {
             return Variable(result, {input}, grad_func);
         }
 
+        Variable relu(const Variable &input)
+        {
+	    auto result = max(input.array(), constant(0, input.array().dims()));
+	    auto grad_func = [](std::vector<Variable> &inputs, const Variable &grad_output) {
+	      inputs[0].addGrad(relu(grad_output) > 0);
+            };
+            return Variable(result, {input}, grad_func);
+        }
```

Comment:
You are sending mask as second input for backward pass. You can use that instead of comparing inputs[0] with 0 again